compiler/hash-typecheck/src/ops/pats.rs

```diff
@@ -94,28 +94,23 @@ impl<'tc> PatMatcher<'tc> {
 
         // Take the first member as we'll use it for our comparison
         let mut members = member_maps.iter();
-        let (first_map, first_pat_id) = members.next().unwrap();
+        let (first_map, _) = members.next().unwrap();
         let first_binds = first_map.keys().copied().collect::<HashSet<Identifier>>();
 
         // Compute the difference in `name` keys, if there exists a
         // difference then we can immediately report the error and abort...
         //
         // @@ErrorReporting: It would be nice to produce multiple errors here
-        for (member, cur_pat_id) in members {
+        for (member, &cur_pat_id) in members {
             // We want to find the largest member and report that that member doesn't
             // contain the binds...
             let cur_binds = member.keys().copied().collect::<HashSet<Identifier>>();
+            let mut diff = first_binds.symmetric_difference(&cur_binds).peekable();
```

Comment:
Well, you basically implemented symmetric difference using the if guarded difference() calls.. So I thought it would be better to use the already existing function for it.

Reply:
oh I missed the `symmetric_difference` and `difference` use